src/edu/psu/sweng500/team8/gui/CellGUI.java

```diff
@@ -0,0 +1,513 @@
+
+		pencilMarkDisplay[2][2] = new JLabel();
+		pencilMarkDisplay[2][2]
+				.setHorizontalAlignment(javax.swing.SwingConstants.CENTER);
+		pencilMarkDisplay[2][2].setText("9");
+		pencilMarkDisplay[2][2].setMaximumSize(new java.awt.Dimension(
+				NUMBER_SIZE, NUMBER_SIZE));
+		pencilMarkDisplay[2][2].setMinimumSize(new java.awt.Dimension(
+				NUMBER_SIZE, NUMBER_SIZE));
+		pencilMarkDisplay[2][2].setPreferredSize(new java.awt.Dimension(
+				NUMBER_SIZE, NUMBER_SIZE));
+		pencilMarkDisplay[2][2].setVisible(false);
+		
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 2;
+		gridBagConstraints.gridy = 2;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+		// gridBagConstraints.insets = new java.awt.Insets(0, 0, 1, 1);
+		pencilMarkDisplayCell.add(pencilMarkDisplay[2][2], gridBagConstraints);
+	}
+
+
+	private void markGivenCell() {
+		HighlightPainter painter = new DefaultHighlighter.DefaultHighlightPainter(
+				Color.green);
+
+		try {
+			// (JN): What is the significance of 0,3?
+			this.numberInputField.getHighlighter().addHighlight(0, 3, painter);
+		} catch (BadLocationException e) {
+			e.printStackTrace();
+			/* TODO - ignore or handle ? */
+		}
+	}
+
+	private void selectCell(CellGUI selectedCell) {
+
+		selectedCell.numberInputField.setBorder(SELECTED_BORDER);
+
+		selectedCell.numberInputCell.setVisible(true);
+		selectedCell.pencilMarkDisplayCell.setVisible(false);
+		selectedCell.pencilMarkInputCell.setVisible(false);
+	}
+	
+	public void selectCell() {
+		selectCell(this);
+	}
+	
+	public void updateSelectedCellFromHint(int number) {
+		this.numberInputField.setText(Integer.toString(number));
+		selectCell(this);
+	}
+
+	public void unselect() {
+		this.numberInputField.setBorder(DEFAULT_BORDER);
+	}
+
+	private JTextField buildReadOnlyTextField(Cell cell) {
+		JTextField textField = new JTextField();
+		textField.setHorizontalAlignment(javax.swing.JTextField.CENTER);
+		// textField.setBorder(javax.swing.BorderFactory.createLineBorder(new
+		// java.awt.Color(0, 0, 0)));
+		textField.setBorder(DEFAULT_BORDER);
+		textField.setEditable(false);
+		textField.setText(Integer.toString(cell.getNumber()));
+
+		if (ValueType.Given.equals(cell.getType())) {
+			HighlightPainter painter = new DefaultHighlighter.DefaultHighlightPainter(
+					Color.green);
+
+			try {
+				textField.getHighlighter().addHighlight(0, 3, painter);
+			} catch (BadLocationException e) {
+				/* TODO - better error handling */
+				e.printStackTrace();
+			}
+		}
+
+		return textField;
+	}
+
+	private CombinationCell buildPencilMarkInputField(Cell cell) {
+		CombinationCell combinationCell = new CombinationCell(cell,
+				this.gameSession);
+
+		/*
+		 * disable any number used according to the cellConstraints for given
+		 * cell
+		 */
+		for (int usedNumber : this.gameSession.getGameBoard()
+				.getCellConstraints(cell).getUsedNumbers()) {

```

Comment:
agree. wasted pointer vs. readability. I saw your comment on the other pull
request asking for reducing number of lines so I had those here and there.
but if we are going for readability overall, I am totally find with that.

On Sun, Jul 12, 2015 at 10:31 PM, Jeremy Nelson <notifications@github.com>
wrote:

> In src/edu/psu/sweng500/team8/gui/CellGUI.java
> <https://github.com/garnryang/Team8/pull/13#discussion_r34430060>:
>
> > +			}
> > +		}
> > +
> > +		return textField;
> > +	}
> > +
> > +	private CombinationCell buildPencilMarkInputField(Cell cell) {
> > +		CombinationCell combinationCell = new CombinationCell(cell,
> > +				this.gameSession);
> > +
> > +		/*
> > +		 * disable any number used according to the cellConstraints for given
> > +		 * cell
> > +		 */
> > +		for (int usedNumber : this.gameSession.getGameBoard()
> > +				.getCellConstraints(cell).getUsedNumbers()) {
>
> Suggest getting the set of used numbers outside the loop. Makes it a bit
> more readable.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/garnryang/Team8/pull/13/files#r34430060>.
>